dspace-server-webapp/src/test/java/org/dspace/app/rest/eperson/DeleteEPersonSubmitterIT.java

```diff
@@ -0,0 +1,382 @@
+        Version version1 = versioningService.createNewVersion(context, item);
+        Integer version1ID = version1.getID();
+        WorkspaceItem version1WorkspaceItem = workspaceItemService.findByItem(context, version1.getItem());
+        installItemService.installItem(context, version1WorkspaceItem);
+
+        assertDeletionOfEperson(submitter);
+        assertNull(retrieveItemSubmitter(item.getID()));
+
+        Item version1Item = retrieveVersionItem(version1ID);
+        assertNull(retrieveItemSubmitter(version1Item.getID()));
+
+
+        context.setCurrentUser(submitterForVersion1);
+
+        Version version2 = versioningService.createNewVersion(context, item);
+        Integer version2ID = version2.getID();
+        WorkspaceItem version2WorkspaceItem = workspaceItemService.findByItem(context, version2.getItem());
+        installItemService.installItem(context, version2WorkspaceItem);
+        Item version2Item = retrieveVersionItem(version2ID);
+        assertEquals(submitterForVersion1.getID(), retrieveItemSubmitter(version2Item.getID()).getID());
+
+        context.setCurrentUser(submitterForVersion2);
+        Version version3 = versioningService.createNewVersion(context, version2Item);
+        Integer version3ID = version3.getID();
+        assertDeletionOfEperson(submitterForVersion2);
+
+        getClient(token).perform(get("/api/versioning/versions/" + version3ID + "/item"))
+                        .andExpect(status().isNoContent());
+
+
+        // Clean up versions
+        cleanupVersion(version1ID);
+        cleanupVersion(version2ID);
+        cleanupVersion(version3ID);
+
+    }
+
+
+    @Test
+    public void testWorkspaceItemSubmitterDelete() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        Community parent = CommunityBuilder.createCommunity(context).build();
+        Collection collection = CollectionBuilder.createCollection(context, parent)
+                                                 .build();
+
+        WorkspaceItem wsi = WorkspaceItemBuilder.createWorkspaceItem(context, collection)
+                                                .withSubmitter(submitter)
+                                                .withTitle("Test Item")
+                                                .withIssueDate("2019-03-06")
+                                                .withSubject("ExtraEntry")
+                                                .build();
+
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get("/api/submission/workspaceitems/" + wsi.getID()))
+                        .andExpect(status().isOk());
+
+        assertDeletionOfEperson(submitter);
+
+        getClient(token).perform(get("/api/submission/workspaceitems/" + wsi.getID()))
+                        .andExpect(status().isNotFound());
+    }
+
+    @Ignore
+    @Test
+    public void testWorkflowItemSubmitterDelete() throws Exception {
```

Comment:
Minor thing.  This test seems to have an `@Ignore`.  Is there a reason for that?

Also, I think this test may be incorrect.  In `EPersonTest.testCascadingDeleteSubmitterPreservesWorkflowItems()`, the test proves that a WorkflowItem is *kept* even if the Submitter is deleted.  This test seems to be trying to prove the *opposite* as the final test in this method is trying to prove that the WorkflowItem `isNotFound()` after the Submitter is deleted.  I suspect that last line should be changed to `isOk()`, as the WorkflowItem should still exist even if the Submitter is deleted.

Reply:
I've logged the above Ignored test as part of #2994.  It's related to the minor bug that I described there.